WcaOnRails/app/webpacker/lib/requests/routes.js.erb

```diff
@@ -88,17 +88,29 @@ export const adminImportResultsUrl = (competitionId) => `<%= CGI.unescape(Rails.
 
 export const adminCheckUploadedResults = (competitionId) => `<%= CGI.unescape(Rails.application.routes.url_helpers.competition_admin_upload_results_edit_path("${competitionId}"))%>`;
 
-export const delegateProbationDataUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.delegate_probation_data_path(format: "json")) %>`;
-
 export const startDelegateProbationUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.start_delegate_probation_path) %>`;
 
 export const endDelegateProbationUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.end_delegate_probation_path) %>`;
 
-// dummyRoleId is used temporarily because delegates is yet to get a role id
+export const rolesOfUser = (userId, params) => {
+  const searchParams = new URLSearchParams(params);
```

Comment:
I am unhappy about this additional `URLSearchParams`. It introduces logic ("take a key-value dict of options and format them prettily into some generic information that can be appended to the URL") and this file should not contain logic. At least not in a stricter sense. It should only glue strings together, not compile params.
- Why is this `params` even necessary in the first place? In my previous review, I suggested splitting the route into three routes, which you have done. Why are you still keeping query params even after the refactor?
- The `params` parameter is currently not used at all, at least not for `rolesOfUser`.
- If it's really, _really_ necessary to have params and they cannot be "hidden" behind clever API design, then please provide an **explicit** list of query params that are hard-coded into the URL. See for example `roleDataUrl` below as a good example.

Reply:
I've removed it now. The reason why I had it was because I'll be needing them for future APIs like list of senior delegates, list of delegates, etc. Also, since there are many types of roles, I was not sure what all types of params are going to come. So I added the generic way. But as you said, the way done in `roleDataUrl` looks better, will follow that while implementing those APIs.